packages/@aws-cdk/aws-apigatewayv2/lib/common/domain-name.ts

```diff
@@ -55,10 +78,79 @@ export interface DomainNameProps {
    * The custom domain name
    */
   readonly domainName: string;
+
   /**
-   * The ACM certificate for this domain name
+   * DomainNameConfigurations for a domain name.
+   * @link https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigatewayv2-domainname.html#cfn-apigatewayv2-domainname-domainnameconfigurations
+   */
+  readonly domainNameConfigurations: DomainNameConfiguration[];
+
+  /**
+    * The mutual TLS authentication configuration for a custom domain name.
+    * @default - mTLS is not configured.
+    */
+  readonly mtls?: MTLSConfig;
+}
+
+/**
+ * Specifies the configuration for a an API's domain name.
+ */
+export interface DomainNameConfiguration {
+  /**
+   * The reference to an AWS-managed certificate for use by the edge-optimized
+   * endpoint for the domain name. For "EDGE" domain names, the certificate
+   * needs to be in the US East (N. Virginia) region.
    */
   readonly certificate: ICertificate;
+
+  /**
+   * The user-friendly name of the certificate that will be used by the endpoint for this domain name.
+   * @default null
+   */
+  readonly certificateName?: string;
+
+  /**
+    * The type of endpoint for this DomainName.
+    * @default REGIONAL
+    */
+  readonly endpointType?: EndpointType;
+
+  /**
+    * The ARN of the public certificate issued by ACM to validate ownership of your
+    * custom domain. Only required when configuring mutual TLS and using an ACM
+    * imported or private CA certificate ARN as the RegionalCertificateArn.
```

Comment:
Confusing. `RegionalCertificateArn` is not here AFAICT. This needs to be written in CDK terms and just copy-pasted from CFN.